radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -88,29 +88,30 @@
 		private final ConnectableObservable<AtomSubmissionUpdate> updates;
 		private final Completable completable;
 
-		private Result(ConnectableObservable<AtomSubmissionUpdate> updates) {
+		private Result(ConnectableObservable<AtomSubmissionUpdate> updates, Completable completable) {
 			this.updates = updates;
-
-			this.completable = updates.filter(AtomSubmissionUpdate::isComplete)
-				.firstOrError()
-				.flatMapCompletable(update -> {
-					if (update.getState() == AtomSubmissionState.STORED) {
-						return Completable.complete();
-					} else {
-						return Completable.error(new ActionExecutionException(update.getData().toString()));
-					}
-				});
+			this.completable = completable;
 		}
 
 		private Result connect() {
 			this.updates.connect();
 			return this;
 		}
 
+		/**
+		 * A low level interface, returns an a observable of the status of an atom submission as it occurs.
+		 * @return observable of atom submission status
+		 */
 		public Observable<AtomSubmissionUpdate> toObservable() {
 			return updates;
 		}
 
+		/**
+		 * A high level interface, returns completable of successful completion of action execution.
```

Comment:
Not really important - having both a low-level and high-level interface exposed through the same interface seems a bit odd.